### PR TITLE
util: Add tests for extended_priority_queue

### PR DESCRIPTION
**Pull request by changlan**

### CODE REVIEW

File: core/utils/extended_priority_queue_test.cc [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+#include <gtest/gtest.h>
+
+#include "random.h"
+
+using bess::utils::extended_priority_queue;
+
+namespace {
+
+// Tests decrease_key_top()
+TEST(ExtendedPriorityQueueTest, DecreseKeyTop) {
```
Comment (barath) [inline core/utils/extended_priority_queue_test.cc]: DecreseKeyTop -> DecreaseKeyTop

File: core/utils/extended_priority_queue_test.cc [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+  extended_priority_queue<int> queue;
+  queue.push(1);
+  queue.push(10000);
+  queue.push(100);
+  queue.push(1000);
+  EXPECT_TRUE(std::is_heap(queue.container().begin(), queue.container().end()));
+
+  int& top = queue.mutable_top();
```
Comment (barath) [inline core/utils/extended_priority_queue_test.cc]: Maybe add an expect here for what top should be right when you pull it out of the heap?

File: core/utils/extended_priority_queue_test.cc [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+  top = 10;
+  EXPECT_EQ(queue.top(), 10);
+
+  queue.decrease_key_top();
+  EXPECT_EQ(queue.top(), 1000);
+}
+
+// Tests delete_single_element()
+TEST(CuckooMapTest, Delete) {
```
Comment (barath) [inline core/utils/extended_priority_queue_test.cc]: Why is this CuckoMapTest?

File: core/utils/extended_priority_queue_test.cc [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+  extended_priority_queue<int> queue;
+  queue.push(1);
+  queue.push(10000);
+  queue.push(100);
+  queue.push(1000);
+  EXPECT_EQ(queue.size(), 4);
+
+  EXPECT_FALSE(
+      queue.delete_single_element([=](const int& x) { return x == 10; }));
+  EXPECT_EQ(queue.size(), 4);
```
Comment (barath) [inline core/utils/extended_priority_queue_test.cc]: You might add an expect for top() after each deletion, to make sure the right value gets pushed to the top of the heap.

**Comment by barath** [comment]
Thanks for adding these tests!

File: core/utils/extended_priority_queue_test.cc (hunk 1 above)
Comment (changlan) [inline core/utils/extended_priority_queue_test.cc]: Fixed.

File: core/utils/extended_priority_queue_test.cc (hunk 2 above)
Comment (changlan) [inline core/utils/extended_priority_queue_test.cc]: Done.

File: core/utils/extended_priority_queue_test.cc (hunk 3 above)
Comment (changlan) [inline core/utils/extended_priority_queue_test.cc]: Good catch. Fixed.

**Comment by barath** [comment]
photo 188

**Pull request merged**